src/neo/Consensus/ConsensusContext.cs

```diff
@@ -37,6 +37,16 @@ internal class ConsensusContext : IDisposable, ISerializable
         // if this node never heard from validator i, LastSeenMessage[i] will be -1.
         public int[] LastSeenMessage;
 
+        /// <summary>
+        /// Future payloads used when CN are probably lagged
+        /// Verified ConsensusPayloads are cached and the lastest height are stored
+        /// They are possibly used when the nodes syncs to the last height
+        /// </summary>
+        public ConsensusPayload[] FuturePreparationPayloads;
+        public ConsensusPayload[] FutureCommitPayloads;
+        public ConsensusPayload[] FutureChangeViewPayloads;
+        public ConsensusPayload[] FutureRecoveryPayloads;
```

Comment:
Why not use only one array to store all of them?

Reply:
This design is similar with the current one, it is easy to set each of them with `new ConsensusPayload[Validators.Length]`.

If we change to a single array we would need to worry about saving information about each index or check all cached ConsensusPayloads before updating it.